spec/validators/case_details/answers_validator_spec.rb

```diff
@@ -0,0 +1,403 @@
+require 'rails_helper'
+
+RSpec.describe CaseDetails::AnswersValidator, type: :model do
+  subject(:validator) { described_class.new(record) }
+
+  let(:record) { instance_double(Case, crime_application:, errors:) }
+  let(:crime_application) { instance_double(CrimeApplication, not_means_tested?: not_means_tested) }
+  let(:errors) { double(:errors) }
+  let(:not_means_tested) { false }
+
+  describe '#validate' do
+    before { allow(record).to receive_messages(**attributes) }
+
+    context 'when all validations pass' do
+      let(:errors) { double(empty?: true) }
+
+      let(:attributes) do
+        {
+          has_case_concluded: 'no',
+          has_codefendants: 'no',
+          is_client_remanded: 'no',
+          values_at: [1],
+          is_first_court_hearing: 'yes',
+          charges: [double(complete?: true)],
+          codefendants: [double(complete?: true)],
+          first_court_hearing_name: nil,
+        }
+      end
+
+      it 'does not add any errors' do
+        subject.validate
+      end
+    end
+
+    context 'when validation fails' do
+      let(:errors) { double(empty?: false) }
+
+      let(:attributes) do
+        {
+          has_case_concluded: 'yes',
+          date_case_concluded: nil,
+          is_preorder_work_claimed: 'yes',
+          is_client_remanded: nil,
+          charges: [double(complete?: false)],
+          has_codefendants: 'yes',
+          codefendants: [double(complete?: false)],
+          is_first_court_hearing: 'no',
+          first_court_hearing_name: nil,
+          values_at: [nil]
+        }
+      end
+
+      it 'adds errors for all failed validations' do # rubocop:disable Spec/MultipleExpectations
+        expect(errors).to receive(:add).with(:has_case_concluded, :blank)
+        expect(errors).to receive(:add).with(:is_preorder_work_claimed, :blank)
+        expect(errors).to receive(:add).with(:is_client_remanded, :blank)
+        expect(errors).to receive(:add).with(:charges, :blank)
+        expect(errors).to receive(:add).with(:charges_summary, :incomplete_records)
+        expect(errors).to receive(:add).with(:codefendants_summary, :incomplete_records)
+        expect(errors).to receive(:add).with(:hearing_details, :blank)
+        expect(errors).to receive(:add).with(:first_court_hearing, :blank)
+
+        expect(errors).to receive(:add).with(:base, :incomplete_records)
+
+        subject.validate
+      end
+
+      context 'when application is not means tested' do
+        let(:not_means_tested) { true }
+
+        it 'adds errors for all failed validations' do
```

Comment:
Is it worth adding `no_to` assertions for the ones that shouldn't be checked? That way if there's a regression the test would fail

Reply:
It blows up on unexpected message if any error not listed is added, so not necessary.